### ISSUE TITLE
Number overlay too large on HiDPI screen

### ISSUE THREAD

**Issue by michael-rapp** [micheleg/dash-to-dock#481]
I just noticed, that the number overlays, which are shown when launching favorite applications via Super + Number key, seem to be too large on HiDPI screens with a window scale factor greater than 1 (see screenshot).

![screenshot from 2017-04-22 19-07-52](https://cloud.githubusercontent.com/assets/6638695/25306554/bb740ad0-278f-11e7-83e4-24c71efb8b58.png)

My system runs Gnome 3.22.2 on Arch Linux. Dash to Dock version is 57.r4.g85f6b8b. Screen resolution is 2880x1920 with a scale factor of 2.

**Comment by micheleg** [comment]
Could you confirm if this is under Wayland or X? Do you have only one or more monitor set to HiDPI?

**Comment by michael-rapp** [comment]
It occurs on Wayland and X. I have only one HiDPI monitor.

**Comment by micheleg** [comment]
I can't seem to be able to reproduce the bug. To factor out some possibilities, if you are using a custom theme could you check if the issue persists with the default theme too?

**Comment by michael-rapp** [comment]
I am using the default Adwaita shell theme. But the "use built-in theme" option is not enabled in the extension's settings.

**Comment by franglais125** [comment]
I managed to reproduce this by setting the HiDPI scaling to 2, and then restarting the session (reloading the extension was not enough). So either Alt+F2 "r" on X, or logging out on Wayland.

![screenshot from 2017-04-29 01-39-24](https://cloud.githubusercontent.com/assets/19195975/25553404/1f3719aa-2c80-11e7-8654-9fd38965da44.png)

I prepared a small fix for this [1]. However, while this seems to fix the issue for the number overlay itself, it would perhaps be better to check what other areas need adjustment (not sure). Also, I don't know what happens if there are many monitors with varying DPIs. I can't test this as my monitors are all low DPI!

[1] https://github.com/franglais125/dash-to-dock/tree/HiDPI_overlay this commit [17a770084ac6aba47ab423197ef717684427bc30]

**Comment by michael-rapp** [comment]
I cannot see any more HiDPI-related issues at the moment. It works quite well. Different scaling factors for different monitors is a thing that isn't really functional in Gnome shell at all. Therefore I avoid to use such a setup.

**Comment by michael-rapp** [comment]
I tested your patch and I can confirm that the issue is gone.

I also tried to use different scaling factors for both of my monitors, but I didn't get it to work properly. If I set a scaling factor (1 or 2) in Gnome Tweak Tool, it is applied to both monitors. If I reset the value to the default (0) in dconf editor, the Gnome shell's UI elements are scaled on both monitors using the factor 2 (I guess, because the primary monitor is the HiDPI one), including the dash-to-dock extension's UI. I don't think that any modifications in the extension are necessary until the shell behaves properly.
